controllers/drplacementcontrol_controller.go

```diff
@@ -1264,9 +1264,53 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("VRG adoption in progress")
 	}
 
-	// delete manifestworks (VRGs)
+	// delete manifestworks (VRGs and NSs)
 	for _, drClusterName := range rmnutil.DRPolicyClusterNames(drPolicy) {
-		err := mwu.DeleteManifestWorksForCluster(drClusterName)
+		// delete VRG manifestwork
+		mwName := mwu.BuildManifestWorkName(rmnutil.MWTypeVRG)
+		mwNamespace := drClusterName
+
+		err = mwu.DeleteManifestWork(mwName, mwNamespace)
+		if err != nil {
+			return fmt.Errorf("%w", err)
+		}
+
+		// check if the Spec has deleteOption. If not call createOrUpdate() to get
+		// the deleteOption added.
+
+		// delete NS manifestwork
+		mwName = mwu.BuildManifestWorkName(rmnutil.MWTypeNS)
+		mwNamespace = drClusterName
+
+		mw := &ocmworkv1.ManifestWork{}
+		err := mwu.Client.Get(mwu.Ctx, types.NamespacedName{Name: mwName, Namespace: mwNamespace}, mw)
+		
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+
+			return fmt.Errorf("failed to retrieve manifestwork for type: %s. Error: %w", mwName, err)
+		}
+
+		// check if the manifestwork has delete Option set
+		// if not set, call CreateOrUpdateNamespaceManifest such that it is
+		// updated with the delete option
+		if mw.Spec.DeleteOption == nil {
+			annotations := make(map[string]string)
+
+			annotations[DRPCNameAnnotation] = drpc.Name
+			annotations[DRPCNamespaceAnnotation] = drpc.Namespace
+
+			err = mwu.CreateOrUpdateNamespaceManifest(drpc.Name, vrgNamespace, drClusterName, annotations)
+			if err != nil {
+				log.Info("error creating namespace via ManifestWork", "error", err, "cluster", drClusterName)
+
+				return err
+			}
+		}
+		
+		err = mwu.DeleteManifestWork(mwName, mwNamespace)
```

Comment:
If we ended up calling `CreateOrUpdateNamespaceManifest` above, this delete may fail stating we have  a stale resource, which is fine as the next reconcile will pick it up and continue, but just noting the observation for now.